frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/modals/attach-deployment-obc-modal.tsx

```diff
@@ -25,20 +25,21 @@ import {
   k8sPatch,
   K8sResourceKind,
   referenceFor,
+  DeploymentKind,
 } from '@console/internal/module/k8s/';
 import { getName } from '@console/shared';
 
 const AttachDeploymentToOBCModal = withHandlePromise((props: AttachDeploymentToOBCModalProps) => {
   const { t } = useTranslation();
-  const [requestDeployment, setRequestedDeployment] = React.useState({});
+  const [requestDeployment, setRequestedDeployment] = React.useState<DeploymentKind>();
   const [deploymentObjects, setDeployments] = React.useState({});
   const [deploymentNames, setDeploymentNames] = React.useState({});
   const { handlePromise, close, cancel, resource, deployments } = props;
 
   const obcName = getName(resource);
-  const deploymentData = _.get(deployments, 'data');
-  const inProgress = _.get(props, 'loaded');
-  const errorMessage = _.get(props, 'loadError');
+  const deploymentData = deployments?.data;
+  const inProgress = props?.inProgress;
```

Comment:
```suggestion
  const inProgress = props?.loaded;
```

Reply:
@afreen23 I avoided this because it has type issues.

Reply:
Please fix the type issues, its also one of the reason for moving to optional chaining instead of `_.get`

Reply:
There are no props coming in for `loaded` and `loadError` rather its on `FirehoseResult`.
So looks like what you did to fix it here is actually a bug fix for this.
So, let's move these to L37 where other props are getting destructured.

Reply:
@afreen23 Yeah! Understood.